app/src/main/java/fr/free/nrw/commons/nearby/NearbyMapFragment.java

```diff
@@ -73,6 +108,114 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         return mapView;
     }
 
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        this.getView().setFocusableInTouchMode(true);
+        this.getView().requestFocus();
+        this.getView().setOnKeyListener( new View.OnKeyListener() {
+
+            @Override
+            public boolean onKey( View v, int keyCode, KeyEvent event) {
+                if (keyCode == KeyEvent.KEYCODE_BACK) {
+                    if(bottomSheetDetailsBehavior.getState() == BottomSheetBehavior
+                            .STATE_EXPANDED) {
+                        bottomSheetDetailsBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
+                        return true;
+                    }
+                    else if (bottomSheetDetailsBehavior.getState() == BottomSheetBehavior
+                            .STATE_COLLAPSED) {
+                        bottomSheetDetailsBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
+                        return true;
+                    }
+                }
+                return false;
+            }
+        } );
+    }
+
+    private void initViews() {
+        bottomSheetList = getActivity().findViewById(R.id.bottom_sheet);
```

Comment:
Use butterknife if possible for view bindings

Reply:
Yes, it was a TODO in my mind. Thanks for remind! @maskaravivek

Reply:
Actually I don't think we can bind views from parent activity with ButterKnife.